content/learn/book/migration-guides/0.8-0.9/_index.md

```diff
@@ -0,0 +1,825 @@
+
+```rust
+// Old (Bevy 0.8)
+app
+  .insert_resource(ImageSettings::default_nearest())
+  .add_plugins(DefaultPlugins)
+
+// New (Bevy 0.9)
+app.add_plugins(DefaultPlugins.set(ImagePlugin::default_nearest()))
+```
+
+The `DefaultTaskPoolOptions` settings have been moved from a resource to `CorePlugin::task_pool_options`:
+
+```rust
+// Old (Bevy 0.8)
+app
+  .insert_resource(DefaultTaskPoolOptions::with_num_threads(4))
+  .add_plugins(DefaultPlugins)
+
+// New (Bevy 0.9)
+app.add_plugins(DefaultPlugins.set(CorePlugin {
+  task_pool_options: TaskPoolOptions::with_num_threads(4),
+}))
+```
+
+### [Remove `AssetServer::watch_for_changes()`](https://github.com/bevyengine/bevy/pull/5968)
+
+`AssetServer::watch_for_changes()` was removed.
+Instead, set it directly on the `AssetPlugin`.
+
+```rust
+app
+  .add_plugin(DefaultPlugins.set(AssetPlugin {
+    watch_for_changes: true,
+    ..default()
+  }))
+```
+
+### [Spawn now takes a Bundle](https://github.com/bevyengine/bevy/pull/6054)
+
+```rust
+// Old (0.8):
+commands
+  .spawn()
+  .insert_bundle((A, B, C));
+// New (0.9)
+commands.spawn((A, B, C));
+
+// Old (0.8):
+commands.spawn_bundle((A, B, C));
+// New (0.9)
+commands.spawn((A, B, C));
+
+// Old (0.8):
+let entity = commands.spawn().id();
+// New (0.9)
+let entity = commands.spawn_empty().id();
+
+// Old (0.8)
+let entity = world.spawn().id();
+// New (0.9)
+let entity = world.spawn_empty();
+```
+
+### [Accept Bundles for insert and remove. Deprecate `insert`/`remove_bundle`](https://github.com/bevyengine/bevy/pull/6039)
+
+Replace `insert_bundle` with `insert`:
+
+```rust
+// Old (0.8)
+commands.spawn().insert_bundle(SomeBundle::default());
+// New (0.9)
+commands.spawn_empty().insert(SomeBundle::default());
+```
+
+Replace `remove_bundle` with `remove`:
+
+```rust
+// Old (0.8)
+commands.entity(some_entity).remove_bundle::<SomeBundle>();
+// New (0.9)
+commands.entity(some_entity).remove::<SomeBundle>();
+```
+
+Replace `remove_bundle_intersection` with `remove_intersection`:
+
+```rust
+// Old (0.8)
+world.entity_mut(some_entity).remove_bundle_intersection::<SomeBundle>();
+// New (0.9)
+world.entity_mut(some_entity).remove_intersection::<SomeBundle>();
+```
+
+Consider consolidating as many operations as possible to improve ergonomics and cut down on archetype moves:
+
+```rust
+// Old (0.8)
+commands.spawn()
+  .insert_bundle(SomeBundle::default())
+  .insert(SomeComponent);
+
+// New (0.9) - Option 1
+commands.spawn_empty().insert((
+  SomeBundle::default(),
+  SomeComponent,
+))
+
+// New (0.9) - Option 2
+commands.spawn((
+  SomeBundle::default(),
+  SomeComponent,
+))
+```
+
+### [Implement `Bundle` for `Component`. Use `Bundle` tuples for insertion](https://github.com/bevyengine/bevy/pull/2975)
+
+In `derive(Bundle)`, the `bundle` attribute has been removed. Nested bundles are now collapsed automatically. You should remove `#[bundle]` attributes.

```

Comment:
~~~suggestion
The `#[bundle]` attribute is no longer required when deriving `Bundle` for nested bundles.

```rust
#[derive(Bundle)]
struct PlayerBundle {
    #[bundle] // Remove this line
    sprite_bundle: SpriteBundle,
    collider: Collider,
}
```
~~~

I feel like there's a bit of room for confusion here and a code block would make it super clear.